src/newrelic.coffee

```
@@ -72,10 +72,14 @@ plugin = (robot) ->
   _request = (path, cb) ->
     robot.http(apiBaseUrl + path)
       .header('X-Api-Key', apiKey)
-      .header("Content-Type","application/x-www-form-urlencoded")
-
-  get = (path, cb) -> _request(path).get() _parse_response(cb)
-  post = (path, data, cb) -> _request(path).post(data) _parse_response(cb)
+      .header('Content-Type','application/x-www-form-urlencoded')
+
+  get = (path, cb) ->
+    return cb({message:'HUBOT_NEWRELIC_API_KEY is not set'}) if !apiKey
```

I would prefer throw an error object `throw new Error('HUBOT_NEWRELIC_API_KEY is not set')` earlier during plugin initialization.

---

Fixed in master